nemo/collections/duplex_s2s/models/duplex_s2s_model_speech_decoder.py

```diff
@@ -0,0 +1,1083 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import builtins
+import warnings
+from collections import defaultdict
+from contextlib import contextmanager
+from itertools import chain
+from pathlib import Path
+from typing import Any
+from lightning.pytorch.utilities import rank_zero_only
+import hydra
+import sacrebleu
+import torch
+from lightning import LightningModule
+from omegaconf import open_dict
+from torch import Tensor
+from torch.distributed import init_device_mesh
+import torch.distributed as dist
+from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import checkpoint_wrapper
+from torch.distributed.fsdp import MixedPrecisionPolicy, fully_shard, FullyShardedDataParallel
+from torch.distributed.tensor import Replicate, Shard
+from torch.distributed.tensor.parallel import (
+    ColwiseParallel,
+    PrepareModuleInput,
+    RowwiseParallel,
+    SequenceParallel,
+    loss_parallel,
+    parallelize_module,
+)
+from transformers import AutoModel, AutoModelForCausalLM, DynamicCache
+from whisper_normalizer.english import EnglishTextNormalizer
+
+from nemo.collections.asr.models import ASRModel
+from nemo.collections.common.parts.optional_cuda_graphs import WithOptionalCudaGraphs
+from nemo.collections.common.tokenizers import AutoTokenizer
+from nemo.collections.duplex_s2s.modules import AudioPerceptionModule
+from nemo.collections.duplex_s2s.modules import TransformerARSpeechDecoder
+from nemo.collections.tts.models import AudioCodecModel
+from nemo.core.neural_types import AudioSignal, LabelsType, LengthsType, NeuralType
+from nemo.utils import logging
+
+import torch.distributed as dist
+
+
+class DuplexS2SModel(LightningModule):
```

Comment:
I think we should remove this class and add its methods to the DuplexS2SModelSpeechDecoder class.